snapcraft/plugins/gradle.py

```diff
@@ -110,3 +119,27 @@ def _get_proxy_options(self):
                     proxy_options.append(
                         '-D{}.proxyPort={}'.format(var, parsed_url.port))
         return proxy_options
+
+
+def _link_or_copy(source, destination, boundary):
```

Comment:
it would be nice to extend `fileutils.link_or_copy` with this logic. Maybe a new parameter `copy_out_of_boundary` (set to `False` by default to not break compatibility.